crates/data_components/src/databricks/sql_warehouse.rs

```diff
@@ -389,6 +410,24 @@ fn map_databricks_type(type_name: &str) -> Result<DataType, Error> {
         return Ok(DataType::Decimal128(precision, scale));
     }
 
+    if type_name_upper.starts_with("STRUCT") {
```

Comment:
I think it would be worth it to define a mini lexer grammar and use a library to parse this instead of doing it manually. See https://github.com/spiceai/spiceai/blob/trunk/crates/runtime/src/secrets/lexer.rs for an example.

Reply:
https://docs.rs/logos/latest/logos/